sdv/model_generator/tree_generator/file_import.py

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2023 Robert Bosch GmbH
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import re
+from typing import List
+
+from sdv.model_generator.tree_generator.constants import JSON, VSPEC
+from sdv.model_generator.tree_generator.file_formats import Json, Vspec, formats
+
+
+# if no other file supported format is found
+class UnsupportedFileFormat(Exception):
+    def __init__(self, *args, **kwargs):
```

Comment:
Be explicit about the parameters you are expecting. Change `__init__` signature to accept the parameters that you are extracting from `*args` instead of relying on them to be present. You can then throw them into a list and throw them into `Exception.__init__`